src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -113,7 +113,7 @@ public interface IMessageDispatcher : IHostEnvironment
     /// AddListener/RemoveListener methods, and exposes the <see cref="ProgressReporting.ProgressTracker"/> to
     /// query progress.
     /// </summary>
-    public abstract class HostEnvironmentBase<TEnv> : ChannelProviderBase, IHostEnvironment, IDisposable, IChannelProvider, IMessageDispatcher
+    public abstract class HostEnvironmentBase<TEnv> : ChannelProviderBase, IHostEnvironment, IChannelProvider, IMessageDispatcher
```

Comment:
I think we should get rid of the `Dispose` method as well.
It feels like we should fix the whole `IDisposable` concern in 1 change - i.e. remove the Dispose method, remove the "temp file" capability, etc.

Here, we are hanging in limbo where this type doesn't implement IDisposable, but it has a Dispose method, and it holds onto Disposable objects (the IFileHandle instances).